lib/internal/process/execution.js

```diff
@@ -119,7 +121,12 @@ function evalScript(name, body, breakFirstLine, print, shouldLoadESM = false) {
     });
     if (print) {
       const { log } = require('internal/console/global');
-      log(result);
+
+      if (isPromise(result)) {
+        PromisePrototypeThen(result, (resultValue) => log(resultValue));
```

Comment:
You should log some `Promise<${resultValue}>`. We should keep indicating that the result was a promise

Reply:
This I worry about a little… people who write shell scripts and rely on the result shouldn't have to unwrap the computation. e.g.

```sh
VAL = $(node --print 'Promise.resolve(1+1)')
((VAL=VAL+1)
echo "${VAL}"
```

Would not involve a whole bunch of funny grep statements 😢 If the caller wants to add `Promise<>` around the output then I feel like they can probably just do that manually themselves?

Reply:
Then can still use `-e 'promise.then(console.log)`

Reply:
Is this better or worse than something akin to:

```
-e 'promise.then((x) => console.log(`Promise<${x}>`))'
```

I guess I don't know what the common use case is… if people use this for just inspecting stuff then I think defaulting to `Promise<x>` is more useful since it is more descriptive… but if people use this for doing scripting more often than inspecting I think `x` is just more functionally useful.

Reply:
It might be worth adding a flag (e.g. `--await`) to cover that use-case, but that should be a separate PR.

Reply:
> We should keep indicating that the result was a promise

There's no value in knowing if the value is a promise while using `eval` or `print`, it makes sense when printing to console but otherwise people who use these two features want the result only.

> It might be worth adding a flag (e.g. --await) to cover that use-case, but that should be a separate PR.

I don't think introducing a new flag is worthy, the behavior here is very intuitive, just wait for the promise to resolve, adding a flag just introduces complexity in something that should be simple.

Reply:
I think it all go back to what is the end goal.

Reply:
`--print` is just a convenience flag, to save the typing of something like `--input-type=module --eval 'console.log(await somethingAsync())'`. I don’t think it should log intermediate results like `Promise`; if the user wants that, they can use `--eval` to get more detailed and explicit output.

Reply:
> There's no value in knowing if the value is a promise

I completely disagree, there's definitely value in knowing if the value is a promise.